base/test.jl

```diff
@@ -22,6 +22,29 @@ export GenericString
 
 #-----------------------------------------------------------------------
 
+# Backtrace utility functions
+function ip_matches_name(ip, name::String)
+    for fr in StackTraces.lookup(ip)
+        if fr === StackTraces.UNKNOWN || fr.from_c
+            return false
+        end
+        string(fr.file) == name && return true
+    end
+    return false
+end
+
+function scrub_backtrace(bt)
+    do_test_ind = findfirst(addr->Base.REPL.ip_matches_func(addr, :do_test), bt)
+    if do_test_ind != 0 && length(bt) > do_test_ind
+        bt = bt[do_test_ind + 1:end]
+    end
+    name_ind = findfirst(addr->ip_matches_name(addr, "./test.jl"), bt)
```

Comment:
Yes, this would have failed on windows. I (hopefully) made it more OS independent now. Also added a test (that fails on master) so let's see what the AV says.